File: src/2025-02-23-deepclaude-try/javascript/exercises/practice/bottle-song/bottle-song.js
//
// This is only a SKELETON file for the 'Bottle Song' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const recite = (initialBottlesCount, takeDownCount) => {
  const numberWords = {
    10: 'Ten',
    9: 'Nine',
    8: 'Eight',
    7: 'Seven',
    6: 'Six',
    5: 'Five',
    4: 'Four',
    3: 'Three',
    2: 'Two',
    1: 'One',
    0: 'No'
  };

  const verses = [];
  for (let i = 0; i < takeDownCount; i++) {
    const current = initialBottlesCount - i;
    if (current < 1) break;

    const currentWord = numberWords[current];
    const currentBottle = current === 1 ? 'bottle' : 'bottles';
    const nextBottles = current - 1;
    const nextWord = nextBottles === 0 ? 'no' : numberWords[nextBottles].toLowerCase();
    const nextBottle = nextBottles === 1 ? 'bottle' : 'bottles';

    const verse = [
      `${currentWord} green ${currentBottle} hanging on the wall,`,
      `${currentWord} green ${currentBottle} hanging on the wall,`,
      `And if one green bottle should accidentally fall,`,
      `There'll be ${nextWord} green ${nextBottle} hanging on the wall.`
    ].join('\n');

    verses.push(verse);
  }

  return verses.join('\n\n');
};

File: src/2025-03-16-12-54-49--a-deepclaude-full-architect-python-0316-2/python/exercises/practice/bottle-song/bottle_song.py
def recite(start, take=1):
    verses = []
    number_words = {
        10: "Ten", 9: "Nine", 8: "Eight", 7: "Seven", 6: "Six",
        5: "Five", 4: "Four", 3: "Three", 2: "Two", 1: "One", 0: "No"
    }
    
    for i in range(take):
        current_num = start - i
        next_num = current_num - 1
        
        if current_num < 1:
            break
            
        current_word = number_words[current_num]
        next_word = number_words[next_num]
        
        current_bottle = "bottle" if current_num == 1 else "bottles"
        next_bottle = "bottle" if next_num == 1 else "bottles"
        
        verse = [
            f"{current_word} green {current_bottle} hanging on the wall,",
            f"{current_word} green {current_bottle} hanging on the wall,",
            f"And if one green bottle should accidentally fall,",
            f"There'll be {next_word.lower()} green {next_bottle} hanging on the wall."
        ]
        
        verses.extend(verse)
        if i < take - 1 and current_num > 1:
            verses.append("")
    
    return verses
